noma/bitcoind.py

```diff
@@ -212,99 +221,92 @@ def generate_rpcauth(username, password=""):
-    bitcoind_dir_exists = pathlib.Path(bitcoind_dir).is_dir()
-
-    if bitcoind_dir_exists:
-        print("bitcoind directory exists")
+    if cfg.BITCOIN_PATH.is_dir():
+        print("✅ bitcoind directory exists")
     else:
-        print("bitcoin folder missing")
-
-    bitcoind_conf = "/media/archive/archive/bitcoin/bitcoin.conf"
-    bitcoind_conf_exists = pathlib.Path(bitcoind_conf).is_file()
+        raise IOError("❌ bitcoin directory missing")
 
-    if bitcoind_conf_exists:
-        print("bitcoin.conf exists")
+    if cfg.BITCOIN_CONF.is_file():
+        print("✅ bitcoind config file exists")
     else:
-        print("bitcoin.conf missing")
+        raise IOError("❌ bitcoin config file is missing")
+    return True
+
 
-    if bitcoind_conf_exists and bitcoind_dir_exists:
-        return True
-    return False
+class IniConfig(ConfigObj):
+    """Allow ini-style ; comments"""
+    COMMENT_MARKERS = ['#', ';']
 
 
-def get_kv(key, config_path):
+def get_ini_kv(config_path, key, section):
     """
-    Parse key-value config files and print out values
+    Parse ini-style config files and print out values
 
     :param key: left part of key value pair
     :param config_path: path to file
     :return: value of key
     """
-    import itertools
-    import configparser
-
-    parser = configparser.ConfigParser(strict=False)
-    with open(config_path) as lines:
-        lines = itertools.chain(
-            ("[main]",), lines
-        )  # workaround: prepend dummy section
-        parser.read_file(lines)
-        return parser.get("main", key)
+    try:
+        config = IniConfig(config_path, file_error=True, encoding='utf-8')
+    except SyntaxError as error:
+        print("⚠️ " + str(error))
+    else:
+        if section:
+            return config[section][key]
+        try:
+            value = config[key]
+        except KeyError:
+            print("❌ key '{k}' does not exist".format(k=key))
+        else:
+            return value
 
 
-def set_kv(key, value, config_path):
+def set_ini_kv(key, value, config_path, section):
     """
     Set key to value in path
     kv pairs are separated by "="
 
     :param key: key to set
     :param value: value to set
     :param config_path: config file path
-    :return str: string written
+    :param section: config section
     """
-    from fileinput import FileInput
+    try:
+        config = IniConfig(config_path, file_error=True, encoding='utf-8')
+    except SyntaxError as error:
+        print("⚠️ " + str(error))
+    else:
+        if section:
+            config[section][key] = value
+        else:
+            config[key] = value
+        config.write(delimiter='=')
 
-    path = pathlib.Path(config_path)
-    config_exists = path.is_file()
 
-    if not config_exists:
-        # create empty config file
-        path.touch()
+def get_toml_kv(config_path, key, section=""):
+    """get toml key value"""
+    config = toml.load(config_path)
+    if section:
```

Comment:
```python
if section:
    return config[section][key]

return config[key]
```

:P

Reply:
Or even (if possible in Python):

```python
if section:
	config = config[section]

return config[key]
```

Reply:
```
if section in config:
    return config[section][key]
```
is correct python, if it's workable code is left up the reader :stuck_out_tongue:

Reply:
Sure.  Most of my comments are suggestions.  Focused on improving: style, readability, ease of maintenance, and error avoidance as the code changes and evolves.

And for that reason my reviews aren't "request changes", but rather just comments, that are left to the PR submitter to decide whether it's useful and they agree, or not 🤷🏻‍♂️.